crates/wasmer-provider/src/pipes.rs

```diff
@@ -0,0 +1,102 @@
+use std::{
+    convert::TryFrom,
+    fmt::Debug,
+    io::{self, Read, Seek, Write},
+};
+
+use anyhow::Error;
+use serde::{Deserialize, Serialize};
+use tokio::fs::File;
+use wasmer_wasi::{VirtualFile, WasiFsError};
+
+/// For piping stdio. Stores all output / input in a byte-vector.
+pub struct FilePipe {
```

Comment:
Out of curiosity, is there a reason not to use the newtype pattern? i.e.

```
pub struct FilePipe(std::fs::File);
```

https://doc.rust-lang.org/book/ch19-04-advanced-types.html#using-the-newtype-pattern-for-type-safety-and-abstraction